modules/addons/aws/external-dns/main.tf

```diff
@@ -17,63 +38,111 @@ resource "aws_route53_record" "ns" {
+            }
+        ]
+    }
+POLICY
+}
+
+resource "aws_iam_role_policy_attachment" "policy_attachment_service_account" {
+  policy_arn = aws_iam_policy.policy_service_account.arn
+  role       = aws_iam_role.role_service_account.name
+}
+
+resource "kubernetes_namespace" "istio_system" {
+  metadata {
+    name = var.service_account_namespace
   }
+  # lifecycle {
+  #   prevent_destroy = true
+  # }
 }
 
-module "external_dns_helm" {
-  source                   = "lablabs/eks-external-dns/aws"
-  argo_enabled             = false
-  argo_helm_enabled        = false
-  irsa_assume_role_enabled = false
-  irsa_role_name_prefix    = var.cluster_name
-
-  cluster_identity_oidc_issuer     = var.cluster_oidc_issuer_url
-  cluster_identity_oidc_issuer_arn = var.oidc_provider_arn
-  irsa_tags                        = var.tags
-
-  helm_release_name     = "external-dns"
-  helm_repo_url         = "https://charts.bitnami.com/bitnami"
-  helm_chart_name       = "external-dns"
-  helm_create_namespace = true
-  namespace             = "external-dns"
-  helm_timeout          = "900"
-  helm_wait             = true
-
-  values = yamlencode({
-    "policy" : "upsert-only"
-    "registry" : "txt"
-    "txtOwnerId" : var.cluster_name
-    "domainFilters" : [aws_route53_zone.cluster.name]
-    "sources" : [var.sources]
-    "annotationFilter" : var.annotation_filter
-    "labelFilter" : var.label_filter
-    "provider" : "aws"
-    "interval" : var.interval
-  })
+
+resource "kubernetes_service_account" "service_account" {
+    metadata {
+        name = "${var.service_account_name}"
+        annotations = {
+            "eks.amazonaws.com/role-arn" = aws_iam_role.role_service_account.arn
+        }
+        namespace = var.service_account_namespace
+    }
+
 }
 
 resource "local_file" "aws_cleanup" {
   content = templatefile("${path.module}/external-dns_aws_cleanup.sh.tmpl", {
-    name_prefix = "eks-${regex(".+-", var.name_prefix)}"
-  })
-  filename        = "${var.output_path}/${var.name_prefix}-external-dns-aws-cleanup.sh"
+    cluster_name = "${var.cluster_name}"
+})
+  filename        = "${var.output_path}/${var.cluster_name}-external-dns-aws-cleanup.sh"
   file_permission = "0755"
 }
 
+# resource "local_file" "aws_cleanup" {
+#   content = templatefile("${path.module}/external-dns_aws_cleanup.sh.tmpl", {
+#     name_prefix = "eks-${regex(".+-", var.name_prefix)}"
+#   })
+#   filename        = "${var.output_path}/${var.name_prefix}-external-dns-aws-cleanup.sh"
+#   file_permission = "0755"
+# }
+
 resource "null_resource" "aws_cleanup" {
   triggers = {
     output_path = var.output_path
-    name_prefix = var.name_prefix
+    cluster_name = var.cluster_name
   }
 
   provisioner "local-exec" {
     when       = destroy
-    command    = "sh ${self.triggers.output_path}/${self.triggers.name_prefix}-external-dns-aws-cleanup.sh"
+    command    = "sh ${self.triggers.output_path}/${self.triggers.cluster_name}-external-dns-aws-cleanup.sh"
```

Comment:
external-dns dependencies should be fitted into route53 controller module, technically we should remove external-dns and commit a new route53 controller module so it be correlated easier...